### PR TITLE
added score counter and saving/loading of high score

### PR DESCRIPTION
**Pull request by Triggered41**

### CODE REVIEW

File: HUD.gd [hunk 1]
```diff
@@ -0,0 +1,35 @@
+extends Control
+
+onready var parent = get_parent()
+onready var Score_label = $score_lable
+var current_score = 0
+
+
+func _physics_process(delta):
+	increase_score()
```
Comment (rohitanwar) [inline HUD.gd]: can't you just do 
```
current_score = snake_body.size()
```

Comment (rohitanwar) [reply HUD.gd]: instead of the whole increase_score function

File: Game.gd [hunk 2]
```diff
@@ -9,7 +9,11 @@ var snake_body = [Vector2(5,10),Vector2(4,10),Vector2(3,10)]
 var snake_direction = Vector2(1,0)
 var add_apple = false
 
+var game_over = false
+var apple_ate = false
```
Comment (rohitanwar) [inline Game.gd]:
```suggestion
```

File: Game.gd [hunk 3]
```diff
@@ -130,6 +136,7 @@ func reset():
 func check_apple_eaten():
 	if applepos == snake_body[0]:
 		add_apple = true
+		apple_ate = true
```
Comment (rohitanwar) [inline Game.gd]:
```suggestion
```

File: HUD.gd (hunk 1 above)
Comment (rohitanwar) [inline HUD.gd]:
```suggestion
	current_score = snake_body.size()
```

File: HUD.gd [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+		parent.game_over = false
+		reset_score()
+
+		
+
+func game_over():
+	return parent.game_over
+
+#Increase the score when apple is eaten
+func increase_score():
```
Comment (rohitanwar) [inline HUD.gd]:
```suggestion
```

File: HUD.gd [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	if parent.apple_ate:
```
Comment (rohitanwar) [inline HUD.gd]:
```suggestion
```

File: HUD.gd [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+		parent.apple_ate = false
```
Comment (rohitanwar) [inline HUD.gd]:
```suggestion
```

File: HUD.gd [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+		current_score += 1
```
Comment (rohitanwar) [inline HUD.gd]:
```suggestion
```

**Review by rohitanwar — CHANGES_REQUESTED** [review]

**Comment by rohitanwar** [comment]
nice work 😄

**Comment by Triggered41** [comment]
Thank you

**Pull request merged**